src/rpc/mining.cpp

```diff
@@ -891,68 +891,72 @@ UniValue estimaterawfee(const JSONRPCRequest& request)
-    passbucket.push_back(Pair("leftmempool", round(buckets.pass.leftMempool * 100.0) / 100.0));
-    result.push_back(Pair("pass", passbucket));
-    UniValue failbucket(UniValue::VOBJ);
-    failbucket.push_back(Pair("startrange", round(buckets.fail.start)));
-    failbucket.push_back(Pair("endrange", round(buckets.fail.end)));
-    failbucket.push_back(Pair("withintarget", round(buckets.fail.withinTarget * 100.0) / 100.0));
-    failbucket.push_back(Pair("totalconfirmed", round(buckets.fail.totalConfirmed * 100.0) / 100.0));
-    failbucket.push_back(Pair("inmempool", round(buckets.fail.inMempool * 100.0) / 100.0));
-    failbucket.push_back(Pair("leftmempool", round(buckets.fail.leftMempool * 100.0) / 100.0));
-    result.push_back(Pair("fail", failbucket));
+    const char* horizon_names[] = {"short", "medium", "long"};
+
+    for (FeeEstimateHorizon horizon : {FeeEstimateHorizon::SHORT_HALFLIFE, FeeEstimateHorizon::MED_HALFLIFE, FeeEstimateHorizon::LONG_HALFLIFE}) {
+        CFeeRate feeRate;
+        EstimationResult buckets;
+        feeRate = ::feeEstimator.estimateRawFee(nBlocks, threshold, horizon, &buckets);
+
+        UniValue horizon_result(UniValue::VOBJ);
+        horizon_result.push_back(Pair("feerate", feeRate == CFeeRate(0) ? -1.0 : ValueFromAmount(feeRate.GetFeePerK())));
+        if (!(feeRate == CFeeRate(0))) {
+            horizon_result.push_back(Pair("decay", buckets.decay));
+            horizon_result.push_back(Pair("scale", (int)buckets.scale));
+            UniValue passbucket(UniValue::VOBJ);
+            passbucket.push_back(Pair("startrange", round(buckets.pass.start)));
+            passbucket.push_back(Pair("endrange", round(buckets.pass.end)));
+            passbucket.push_back(Pair("withintarget", round(buckets.pass.withinTarget * 100.0) / 100.0));
+            passbucket.push_back(Pair("totalconfirmed", round(buckets.pass.totalConfirmed * 100.0) / 100.0));
+            passbucket.push_back(Pair("inmempool", round(buckets.pass.inMempool * 100.0) / 100.0));
+            passbucket.push_back(Pair("leftmempool", round(buckets.pass.leftMempool * 100.0) / 100.0));
+            horizon_result.push_back(Pair("pass", passbucket));
+            UniValue failbucket(UniValue::VOBJ);
+            failbucket.push_back(Pair("startrange", round(buckets.fail.start)));
+            failbucket.push_back(Pair("endrange", round(buckets.fail.end)));
+            failbucket.push_back(Pair("withintarget", round(buckets.fail.withinTarget * 100.0) / 100.0));
+            failbucket.push_back(Pair("totalconfirmed", round(buckets.fail.totalConfirmed * 100.0) / 100.0));
+            failbucket.push_back(Pair("inmempool", round(buckets.fail.inMempool * 100.0) / 100.0));
+            failbucket.push_back(Pair("leftmempool", round(buckets.fail.leftMempool * 100.0) / 100.0));
+            horizon_result.push_back(Pair("fail", failbucket));
+        }
+        result.push_back(Pair(horizon_names[horizon], horizon_result));
```

Comment:
In commit "Change API to estimaterawfee"

This is pretty fragile. Consider replacing `horizon_names` array with `FeeEstimateHorizon -> std::string` function, or at adding a comment to FeeEstimateHorizon saying the enum values are used as array indices and need to be stable.